### PR TITLE
New package: python3-language-server-0.31.4

### PR DESCRIPTION
**Pull request by k4leg**

### CODE REVIEW

**Comment by q66** [comment]
No new python2 stuff please

File: srcpkgs/python-language-server/template [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+build_style=python-module
+pycompile_module="pyls"
+short_desc="Python 2 implementation of the Language Server Protocol"
+maintainer="k4leg <d0xi@inbox.ru>"
+license="MIT"
+homepage="https://github.com/palantir/python-language-server"
+distfiles="${homepage}/archive/${version}.tar.gz"
+checksum=3555ebe496858f896eb32c713ce244d5897da2be52a9c17d199ec4026371ad43
+
+if [ "$CROSS_BUILD" ]; then
```
Comment (q66) [inline srcpkgs/python-language-server/template]: This seems wrong

Comment (k4leg) [reply srcpkgs/python-language-server/template]: What exactly? Everything seems to be in order.

Comment (q66) [reply srcpkgs/python-language-server/template]: Even if this was correct, you could just make it hostmakedepends always (when not cross compiling, hostmakedepends and makedepends is the same). But as it seems to me, stuff like setuptools should always be hostmakedepends, while "target" dependencies should be makedepends.

**Comment by q66** [comment]
Rename to python3-language-server

File: srcpkgs/python-language-server/template (hunk 1 above)
Comment (k4leg) [reply srcpkgs/python-language-server/template]: Yes, thank you!

**Comment by q66** [comment]
add `wrksrc` for `python-language-server-${version}`; also verify whether run-time `depends=` is being filled correctly, and manually specify any run-time dependencies that are not automatically added.

File: srcpkgs/python3-jsonrpc-server/template [hunk 2]
```diff
@@ -0,0 +1,20 @@
+# Template file for 'python3-jsonrpc-server'
+pkgname=python3-jsonrpc-server
+version=0.3.2
+revision=1
+wrksrc="${pkgname/3}-${version}"
+build_style=python3-module
+pycompile_module="pyls_jsonrpc"
+hostmakedepends="python3-setuptools"
+depends="python3 python3-future python3-ultrajson"
```
Comment (q66) [inline srcpkgs/python3-jsonrpc-server/template]: don't need to depend on `python3`

Comment (k4leg) [reply srcpkgs/python3-jsonrpc-server/template]: Okay.

**Comment by Chocimier** [comment]
```
$ pyls
Traceback (most recent call last):
  File "/usr/lib/python3.8/site-packages/pkg_resources/__init__.py", line 583, in _build_master
    ws.require(__requires__)
  File "/usr/lib/python3.8/site-packages/pkg_resources/__init__.py", line 900, in require
    needed = self.resolve(parse_requirements(requirements))
  File "/usr/lib/python3.8/site-packages/pkg_resources/__init__.py", line 791, in resolve
    raise VersionConflict(dist, req).with_context(dependent_req)
pkg_resources.ContextualVersionConflict: (python-jsonrpc-server 0+unknown (/usr/lib/python3.8/site-packages), Requirement.parse('python-jsonrpc-server>=0.3.2'), {'python-language-server'})
```

There is no need to set pycompile_module, it is computed when not set.
jsonrpc and language-server looks like noarch.

**Comment by k4leg** [comment]
> ```
> $ pyls
> Traceback (most recent call last):
>   File "/usr/lib/python3.8/site-packages/pkg_resources/__init__.py", line 583, in _build_master
>     ws.require(__requires__)
>   File "/usr/lib/python3.8/site-packages/pkg_resources/__init__.py", line 900, in require
>     needed = self.resolve(parse_requirements(requirements))
>   File "/usr/lib/python3.8/site-packages/pkg_resources/__init__.py", line 791, in resolve
>     raise VersionConflict(dist, req).with_context(dependent_req)
> pkg_resources.ContextualVersionConflict: (python-jsonrpc-server 0+unknown (/usr/lib/python3.8/site-packages), Requirement.parse('python-jsonrpc-server>=0.3.2'), {'python-language-server'})
> ```
> 
> There is no need to set pycompile_module, it is computed when not set.
> jsonrpc and language-server looks like noarch.

Fixed, thanks.

**Comment by k4leg** [comment]
Inside the `python3-ultrajson` package there is no BSD license file, and everything is fine.

**Comment by Chocimier** [comment]
There is license as comment in ujson-1.35/python/ujson.c, some templates eg. moreutils creates license file.
Parso should depend on python. Other don't because their dependencies do.

**Comment by Chocimier** [comment]
Don't forget to exchange patches with upstream.

**Pull request merged**